tests/h/traversal/roots_test.py

```diff
@@ -0,0 +1,197 @@
+class TestAnnotationRoot(object):
+    def test_get_item_fetches_annotation(self, pyramid_request, storage):
+        factory = AnnotationRoot(pyramid_request)
+
+        factory['123']
+        storage.fetch_annotation.assert_called_once_with(pyramid_request.db, '123')
+
+    def test_get_item_returns_annotation_resource(self, pyramid_request, storage):
+        factory = AnnotationRoot(pyramid_request)
+        storage.fetch_annotation.return_value = mock.Mock()
+
+        resource = factory['123']
+        assert isinstance(resource, AnnotationContext)
+
+    def test_get_item_resource_has_right_annotation(self, pyramid_request, storage):
+        factory = AnnotationRoot(pyramid_request)
+        storage.fetch_annotation.return_value = mock.Mock()
+
+        resource = factory['123']
+        assert resource.annotation == storage.fetch_annotation.return_value
+
+    def test_get_item_raises_when_annotation_is_not_found(self, storage, pyramid_request):
+        factory = AnnotationRoot(pyramid_request)
+        storage.fetch_annotation.return_value = None
+
+        with pytest.raises(KeyError):
+            factory['123']
+
+    def test_get_item_has_right_group_service(self, pyramid_request, storage, group_service):
+        factory = AnnotationRoot(pyramid_request)
+        storage.fetch_annotation.return_value = mock.Mock()
+
+        resource = factory['123']
+        assert resource.group_service == group_service
+
+    def test_get_item_has_right_links_service(self, pyramid_request, storage, links_service):
+        factory = AnnotationRoot(pyramid_request)
+        storage.fetch_annotation.return_value = mock.Mock()
+
+        resource = factory['123']
+        assert resource.links_service == links_service
+
+    @pytest.fixture
+    def storage(self, patch):
+        return patch('h.traversal.roots.storage')
+
+    @pytest.fixture
+    def group_service(self, pyramid_config):
+        group_service = mock.Mock(spec_set=['find'])
+        pyramid_config.register_service(group_service, iface='h.interfaces.IGroupService')
+        return group_service
+
+    @pytest.fixture
+    def links_service(self, pyramid_config):
+        service = mock.Mock()
+        pyramid_config.register_service(service, name='links')
+        return service
+
+
+class TestAuthClientResourceFactory(object):
+    def test_get_item_returns_an_authclient(self, pyramid_request):
+        authclient = AuthClient(name='test', authority='example.com')
+        pyramid_request.db.add(authclient)
+        pyramid_request.db.flush()
+
+        factory = AuthClientRoot(pyramid_request)
+        assert factory[authclient.id] == authclient
+
+    def test_get_item_returns_keyerror_if_not_found(self, pyramid_request):
+        factory = AuthClientRoot(pyramid_request)
+        with pytest.raises(KeyError):
+            factory['E19D247D-1F07-4E91-B40D-00DF22E693E4']
+
+    def test_get_item_returns_keyerror_if_invalid(self, pyramid_request):
```

Comment:
The naming of the test methods in this module for operations that raise KeyErrors is inconsistent (case, e.g.).